File: app/src/main/java/syed/com/wordrepo/dao/WordDao.java
package syed.com.wordrepo.dao;

import android.arch.lifecycle.LiveData;
import android.arch.persistence.room.Dao;
import android.arch.persistence.room.Delete;
import android.arch.persistence.room.Insert;
import android.arch.persistence.room.Query;
import android.arch.persistence.room.Update;

import java.util.List;

import syed.com.wordrepo.entitiy.Word;

/**
 * Created by syed on 1/11/18.
 */
@Dao
public interface WordDao {
    @Insert
    void insertWord(Word word);

    @Update
    void updateWord(Word word);

    @Delete
    void deleteWord(Word word);

    @Query("SELECT * FROM word_table ORDER BY word ASC")
    LiveData<List<Word>> getAllWords();
}

File: app/src/main/java/syed/com/wordrepo/dao/WordDao.kt
package syed.com.wordrepo.dao

import android.arch.lifecycle.LiveData
import android.arch.persistence.room.*
import syed.com.wordrepo.entitiy.Word

/**
 * Created by syed on 2/13/18.
 */

@Dao
interface WordDao {
    @Insert
    fun insertWord(word: Word)

    @Update
    fun updateWord(word: Word)

    @Delete
    fun deleteWord(word: Word)

    @Query("SELECT * FROM word_table WHERE id = :arg0")
    fun getWord(id: Int): Word

    @Query("SELECT * FROM word_table ORDER BY word ASC")
    fun getAllWords(): LiveData<List<Word>>

    //Todo make it more generic by passing the table name to the method
    @Query("SELECT COUNT(*) FROM word_table")
    fun getWordTableSize(): Int
}
